bin/bout-v5-input-file-upgrader.py

```diff
@@ -0,0 +1,200 @@
+#!/usr/bin/env python3
+
+import argparse
+import copy
+import difflib
+import textwrap
+import warnings
+
+from boutdata.data import BoutOptionsFile
+from boututils.boutwarnings import AlwaysWarning
+
+# This should be a list of dicts, each containing exactly two keys:
+# "old" and "new". The values of these keys should be the old/new
+# names of input file values or sections
+REPLACEMENTS = [
+    {"old": "mesh:paralleltransform", "new": "mesh:paralleltransform:type"},
+    {"old": "fci", "new": "mesh:paralleltransform"},
+    {"old": "interpolation", "new": "mesh:paralleltransform:xzinterpolation"},
+]
+
+
+def fix_replacements(replacements, options_file):
+    """Change the names of options in options_file according to the list
+    of dicts replacements
+
+    """
+    for replacement in replacements:
+        try:
+            options_file.rename(replacement["old"], replacement["new"])
+        except KeyError:
+            pass
+        except TypeError as e:
+            raise RuntimeError(
+                "Could not apply transformation: '{old}' -> '{new}' to file '{0}', due to error:"
+                "\n\t{1}".format(options_file.filename, e.args[0], **replacement)
+            ) from e
+
+
+def apply_fixes(replacements, options_file):
+    """Apply all fixes in this module
+    """
+
+    modified = copy.deepcopy(options_file)
+
+    fix_replacements(replacements, options_file)
```

Comment:
Should this apply `fix_replacements` to `modified` rather than `options_file`?

Reply:
Good spot, thanks!